src/main/java/org/folio/marc4ld/service/ld2marc/mapper/Ld2MarcMapper.java

```diff
@@ -1,12 +1,11 @@
 package org.folio.marc4ld.service.ld2marc.mapper;
 
-import org.folio.ld.dictionary.PredicateDictionary;
-import org.folio.ld.dictionary.model.Resource;
+import org.folio.ld.dictionary.model.ResourceEdge;
 import org.marc4j.marc.DataField;
 
 public interface Ld2MarcMapper {
 
-  boolean canMap(PredicateDictionary predicate, Resource resource);
+  boolean canMap(ResourceEdge resourceEdge);
 
-  DataField map(Resource resource);
+  DataField map(ResourceEdge resourceEdge);
 }
```

Comment:
minor suggestion for refactoring
```java
public interface Ld2MarcMapper extend Predicate <ResourceEdge>, Function<ResourceEdge, DataField>{

 default Optional<DataField> map(ResourceEdge resourceEdge); {
     return Optional.of(resourceEdge}
               .filter(this::test)
               .map(this::apply);
}
```
old methods:
canMap --> test
map      --> apply

outside the class use ``flatmap( res - > Ld2MarcMapper.map(res)`
instead of caMap --> map

Reply:
done